src/Polly.Core/Simmy/MonkeyStrategy.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Polly.Simmy.Utils;
+
+namespace Polly.Simmy;
+
+#pragma warning disable CA1031 // Do not catch general exception types
+#pragma warning disable S3928 // Custom ArgumentNullException message
+
+/// <summary>
+/// Contains common functionality for chaos strategies which intentionally disrupt executions - which monkey around with calls.
+/// </summary>
+public abstract class MonkeyStrategy : ResilienceStrategy
+{
+    private readonly Func<double> _randomizer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MonkeyStrategy"/> class.
+    /// </summary>
+    /// <param name="options">The chaos strategy options.</param>
+    protected MonkeyStrategy(MonkeyStrategyOptions options)
```

Comment:
Pass `Func<double>` randomizer explicitly?

Reply:
I guess this makes sense if we decide to rely on the `CompositeStrategyBuilderBase.Randomizer`

Reply:
I am thinking of dropping it from the builder. I think your approach is fine. 

I will be doing something similar for `RetryStrategyOptions` - > introduce `RetryStrategyOptions.Randomizer`